sql/sort_field.go

```diff
@@ -0,0 +1,100 @@
+// Copyright 2021 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package sql
+
+import (
+	"fmt"
+
+	"gopkg.in/src-d/go-errors.v1"
+)
+
+// SortField is a field by which a query will be sorted.
+type SortField struct {
+	// Column to order by.
+	Column Expression
+	// Order type.
+	Order SortOrder
+	// NullOrdering defining how nulls will be ordered.
+	NullOrdering NullOrdering
+}
+
+type SortFields []SortField
+
+func (sf SortFields) ToExpressions() []Expression {
+	es := make([]Expression, len(sf))
+	for i, f := range sf {
+		es[i] = f.Column
+	}
+	return es
+}
+
+func (sf SortFields) FromExpressions(exprs []Expression) SortFields {
+	var fields = make(SortFields, len(sf))
```

Comment:
Makes more sense to take the length of `exprs` rather than `sf`? Or we should throw an explicit panic if the lengths don't match.